tests/tests_fabric/strategies/test_xla_fsdp.py

```diff
@@ -0,0 +1,286 @@
+                strategy.setup_optimizer(bad_optimizer_1)
+            with pytest.raises(ValueError, match="The optimizer does not seem to reference any XLA FSDP parameter"):
+                strategy.setup_optimizer(bad_optimizer_2)
+
+
+@RunIf(min_torch="2.0", tpu=True)
+def test_xla_fsdp_no_backward_sync():
+    """Test that the backward sync control calls `.no_sync()`, and only on a module wrapped in
+    XlaFullyShardedDataParallel."""
+
+    strategy = XLAFSDPStrategy()
+    assert isinstance(strategy._backward_sync_control, _XLAFSDPBackwardSyncControl)
+
+    with pytest.raises(
+        TypeError, match="is only possible if the module passed to .* is wrapped in `XlaFullyShardedDataParallel`"
+    ), strategy._backward_sync_control.no_backward_sync(object()):
+        pass
+
+    from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel
+
+    module = MagicMock(spec=XlaFullyShardedDataParallel)
+    with strategy._backward_sync_control.no_backward_sync(module):
+        pass
+
+    module.no_sync.assert_called_once()
+
+
+@RunIf(min_torch="2.0", tpu=True)
+def test_xla_fsdp_grad_clipping_value_error():
+    strategy = XLAFSDPStrategy()
+    with pytest.raises(NotImplementedError, match="does not support to clip gradients by value"):
+        strategy.clip_gradients_value(Mock(), Mock(), Mock())
+
+
+@RunIf(min_torch="2.0", tpu=True)
+def test_xla_fsdp_grad_clipping_norm_error():
+    strategy = XLAFSDPStrategy()
+    with pytest.raises(
+        TypeError,
+        match="only possible if the module.*is wrapped in `XLAFullyShardedDataParallel`",
+    ):
+        strategy.clip_gradients_norm(Mock(), Mock(), Mock())
+
+
+def xla_fsdp_train_save_load(fabric: Fabric, tmp_path, state_dict_type):
+    """Fabric launch function for test_xla_fsdp_train_save_load."""
+    tmp_path = str(tmp_path)
+    device = xm.xla_device()
+    model_1 = torch.nn.Sequential(torch.nn.Linear(32, 32), torch.nn.ReLU(), torch.nn.Linear(32, 2))
+    model_1.to(device)
+    model_1 = fabric.setup_module(model_1)
+
+    optimizer_1 = torch.optim.Adam(model_1.parameters(), lr=0.1)
+    optimizer_1 = fabric.setup_optimizers(optimizer_1)
+
+    dataloader = DataLoader(RandomDataset(32, 64))
+    dataloader = fabric.setup_dataloaders(dataloader)
+
+    def step(model, batch):
+        output = model(batch)
+        return torch.nn.functional.mse_loss(output, torch.ones_like(output))
+
+    model_1.train()
+    data_iter = iter(dataloader)
+    batch = next(data_iter)
+    loss = step(model_1, batch)
+    fabric.backward(loss)
+    optimizer_1.step()
+    optimizer_1.zero_grad()
+    xm.mark_step()
+
+    checkpoint_path = fabric.broadcast(tmp_path)
+    checkpoint_filename = fabric.broadcast(f"{tmp_path}/fsdp-checkpoint")
+    params_before = deepcopy(list(model_1.parameters()))
+
+    state = {
+        "model": model_1,
+        "shard_metadata": model_1._forward_module.get_shard_metadata(),
```

Comment:
Is this optional? Shouldn't this be part of the model state already?

Reply:
Are you suggesting that the entire state is optional? This state is defined based on https://github.com/pytorch/xla/blob/master/docs/fsdp.md?plain=1#L43-L47

Reply:
But this isn't great from an user perspective. What if when we are processing `"model": model_1` we automatically do it for the user by saving `shard_metadata()`?

cc @awaelchli

Reply:
this makes sense, I can make this internal